File: Day49/DSA/Array_Subset.java
Question:- https://www.geeksforgeeks.org/problems/array-subset-of-another-array2317/1?page=1&category=Hash&difficulty=Basic&sortBy=submissions


Code:- 


class Solution {
    public boolean isSubset(int a[], int b[]) {
        // Your code here
        HashMap<Integer,Integer> mp = new HashMap<>();
        
        for(int i : a)
        {
            mp.put(i , mp.getOrDefault(i,0) + 1);
        }
        
        for(int i : b)
        {
            if(mp.containsKey(i) && mp.get(i) > 0)
            {
                mp.put(i , mp.get(i) - 1);
            }
            else{
                return false;
            }
        }
        
        return true;
    }
}

File: Day49/DSA/Array_Subset.cpp
Question:- https://www.geeksforgeeks.org/problems/array-subset-of-another-array2317/1?page=1&category=Hash&difficulty=Basic&sortBy=submissions

Code:- 

class Solution {
  public:
    // Function to check if b is a subset of a
    bool isSubset(vector<int> &a, vector<int> &b) {
        // Your code here
        unordered_map<int,int> mp;
        
        for(auto i : a)
        {
            mp[i]++;
        }
        
        for(auto i : b)
        {
            if(mp[i] > 0)
            {
                mp[i]--;
            }
            else
            {
                return false;
            }
        }
        
        return true;
    }
};